projects/components/src/navigation/navigation-list-component.service.ts

```diff
@@ -26,6 +29,25 @@ export class NavigationListComponentService {
     return updatedItems;
   }
 
+  public resolveNavItemConfigTimeRanges(navItems: NavItemConfig[]): Observable<NavItemConfig[]> {
```

Comment:
Called by `NavigationListComponent` to decorate each navItem with it's time range. It retrieves the time range for each navItem with it's respective path from `getUserSpecifiedTimeRangeForPage`, then adds it to the navItem, and returns them.

Reply:
> Kept these apart for readability and FF backwards compatibility

Not sure that we helped in either respect (I would consider the FF evaluation part of the resolution logic, and thus the responsibility of the resolver), but this is really just a style preference, so your call.